overlay/src/network.rs

```diff
@@ -1,125 +1,274 @@
+#[derive(Serialize, Deserialize, Clone)]
+pub struct TeamInfoRaw {
     pub team_name: String,
-    pub players: Vec<(String, u8)>,
+    pub members: Vec<MemberRaw>,
     pub flag: Option<Vec<u8>>,
 }
 
-impl TeamInfo {
-    pub async fn new(url: &String, tournament_id: u32, team_id: u64, team_color: Color) -> Self {
-        debug!(
+impl TeamInfoRaw {
+    pub async fn new(url: &str, tournament_id: u32, team_id: u64, team_color: Color) -> Self {
+        let client = CLIENT_CELL.get().unwrap();
+        info!(
             "Requesting UWH API for team information for team {}",
             team_id
         );
         let data: Value = serde_json::from_str(
-            &reqwest::get(format!(
-                "https://{}/api/v1/tournaments/{}/teams/{}",
-                url, tournament_id, team_id
-            ))
-            .await
-            .unwrap()
-            .text()
-            .await
-            .unwrap(),
+            &client
+                .get(format!(
+                    "http://{url}/api/admin/get-event-team?legacyEventId={tournament_id}&legacyTeamId={team_id}"
+                ))
+                .send()
+                .await
+                .expect("Coudn't request team data")
+                .text()
+                .await
+                .expect("Coudn't get team data body"),
         )
         .unwrap();
-        let players: Vec<Value> = data["team"]["roster"]
-            .as_array()
-            .map(|x| x.to_vec())
-            .unwrap_or_default();
-        let mut player_list: Vec<(String, u8)> = Vec::new();
-        for player in players {
-            player_list.push((
-                player["name"].as_str().unwrap_or("").to_string(),
-                player["number"].as_u64().unwrap_or(0) as u8,
-            ));
-        }
 
-        Self {
-            team_name: data["team"]["name"]
-                .as_str()
-                .unwrap_or(match team_color {
-                    Color::Black => "Black",
-                    Color::White => "White",
-                })
-                .to_string(),
-            players: player_list,
-            flag: data["team"]["flag_url"]
-                .as_str()
-                .map(|s| reqwest::blocking::get(s).unwrap().bytes().unwrap().to_vec()),
-        }
+        let mut members = Vec::new();
+        futures::future::join_all(
+            data["roster"]
+                .as_array()
+                .map(|x| x.to_vec())
+                .unwrap_or_default()
+                .iter()
+                .map(|member| async {
+                    MemberRaw {
+                        name: member["rosterName"].to_string().trim().to_string(),
+                        number: member["capNumber"].as_u64().map(|e| e as u8),
+                        role: member["roles"]
+                            .as_array()
+                            .and_then(|a| a.iter().map(|v| v.to_string()).find(|v| *v == "Player")),
```

Comment:
I think we actually want to invert this filter. We want to show any roles that aren't "Player" in banners above the images (ex "MAN WHO HYPES" in the demo video sent on fiverr). For simplicity we can just choose the first role that isn't "Player"